src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -45,6 +45,74 @@ impl ScalarFunc {
         }
     }
 
+    #[inline]
+    pub fn oct_int<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &[Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        match self.children[0].eval_int(ctx, row) {
+            Err(_e) => self.oct_string(ctx, row),
```

Comment:
Please stay the same with TiDB implementation.
FYI: https://github.com/pingcap/tidb/blob/master/expression/builtin_string.go#L2507

If it's error, return it.

Reply:
+1